packages/app-builder/src/components/Data/DataModelFlow.tsx

```diff
@@ -0,0 +1,269 @@
+import { type TableModel } from '@app-builder/models/data-model';
+import { CreateTable } from '@app-builder/routes/ressources+/data+/createTable';
+import Dagre from '@dagrejs/dagre';
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import ReactFlow, {
+  applyEdgeChanges,
+  applyNodeChanges,
+  ControlButton,
+  Controls,
+  type Edge,
+  type EdgeChange,
+  type Node,
+  type NodeChange,
+  Panel,
+  ReactFlowProvider,
+  useReactFlow,
+} from 'reactflow';
+import reactflowStyles from 'reactflow/dist/style.css';
+import * as R from 'remeda';
+import { Button, MenuButton } from 'ui-design-system';
+import { Icon } from 'ui-icons';
+
+import {
+  SchemaMenuMenuItem,
+  SchemaMenuMenuPopover,
+  SchemaMenuRoot,
+} from '../Schema/SchemaMenu';
+import { dataI18n } from './data-i18n';
+import {
+  adaptDataModelEdges,
+  defaultDataModelEdgeOptions,
+  type LinkToSingleData,
+  LinkToSingleEdge,
+} from './LinkToSingleEdge';
+import {
+  adaptTableModelNode,
+  TableModelNode,
+  type TableModelNodeData,
+} from './TableModelNode';
+
+type DataModelNodeData = TableModelNodeData;
+
+const nodeTypes = {
+  table_model: TableModelNode,
+};
+
+type DataModelEdgeData = LinkToSingleData;
+
+const edgeTypes = {
+  link_to_single_edge: LinkToSingleEdge,
+};
+
+const useDataModelReactFlow = useReactFlow<
+  DataModelNodeData,
+  DataModelEdgeData
+>;
+
+interface DataModelFlowProps {
+  dataModel: TableModel[];
+  children?: React.ReactNode;
+}
+
+export const dataModelFlowStyles = reactflowStyles;
+
+export function DataModelFlow({ dataModel, children }: DataModelFlowProps) {
+  return (
+    <ReactFlowProvider>
+      <DataModelFlowImpl dataModel={dataModel}>{children}</DataModelFlowImpl>
+    </ReactFlowProvider>
+  );
+}
+
+function DataModelFlowImpl({ dataModel, children }: DataModelFlowProps) {
+  const { t } = useTranslation(dataI18n);
+  const [nodes, setNodes] = React.useState<Array<Node<DataModelNodeData>>>([]);
+  const [edges, setEdges] = React.useState<Array<Edge<DataModelEdgeData>>>([]);
+
+  const onNodesChange = React.useCallback((changes: NodeChange[]) => {
+    const allowedChanges = changes.filter((change) => change.type !== 'remove');
+    setNodes((nds) => applyNodeChanges(allowedChanges, nds));
+  }, []);
+  const onEdgesChange = React.useCallback((changes: EdgeChange[]) => {
+    const allowedChanges = changes.filter((change) => change.type !== 'remove');
+    setEdges((eds) => applyEdgeChanges(allowedChanges, eds));
+  }, []);
```

Comment:
I prevent here the "remove" action (before that, you could remove nodes and edges despite this is not triggering any real action on the org data model)